### PR TITLE
Add export event log button

### LINKED ISSUE #2383: Export event log as JSON from console
#### Summary

We should allow users to export the log of events of an entity as JSON.

Example:
image

#### Why do we need this?

To make debugging easier for both us and customers.

#### What is already there? What do you see now?

Volatile event log in the console window

#### What is missing? What do you want to see?

Ability to export the event log seen in the console

#### How do you propose to implement this?

Have an `export as JSON` button in the event log window. When clicked, it should compose a JSON file that includes the raw JSON data of that stream log and initiate a download of this file. The event log can be sourced from the redux store.

#### Can you do this yourself and submit a Pull Request?

maybe as a fab friday :thinking: assigning @bafonins for now

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This pull request adds button to download the live data.
Closes #2383 

#### Changes

- Added export button and onExport() function

Screenshot 2021-12-21 at 09 16 50

#### Testing

Run e2e tests and manually

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/components/events/index.js [hunk 1]
```diff
@@ -50,6 +51,10 @@ const Events = React.memo(
   }) => {
     const [focus, setFocus] = useState({ eventId: undefined, visible: false })
     const onPause = useCallback(() => onPauseToggle(paused), [onPauseToggle, paused])
+    const onExport = useCallback(() => {
+      const eventLogData = composeDataUri(JSON.stringify(events, undefined, 2))
+      downloadDataUriAsFile(eventLogData, 'tts_live_data.json')
```
Comment (kschiffer) [inline pkg/webui/console/components/events/index.js]:
```suggestion
      downloadDataUriAsFile(eventLogData, `${entityId}_live_data_${Date.now()}.json`)
```

**Review by kschiffer — CHANGES_REQUESTED** [review]
Works well but please fix the responsive styling.

image

The responsive styling has been problematic there for a while already. What would help here is hiding the headers (`Time`, `Type`, `Data`, `Entity`) in mobile views. Then, the buttons can move to the right and should also use the `responsiveLabel` prop.

**Comment by ryaplots** [comment]
After the last changes it would look like this:
Screenshot 2021-12-21 at 14 51 00
Screenshot 2021-12-21 at 14 51 18
Screenshot 2021-12-21 at 14 54 20

**Review by bafonins — APPROVED** [review]

**Review by kschiffer — APPROVED** [review]
Please also apply the following diff:
```udiff
diff --git a/pkg/webui/components/button/button.styl b/pkg/webui/components/button/button.styl
index ea4473ab0..13a61d620 100644
--- a/pkg/webui/components/button/button.styl
+++ b/pkg/webui/components/button/button.styl
@@ -138,6 +138,7 @@ $button($color)
         margin-right: $cs.xs
 
     &.responsive-label
+      padding: $cs.xs
       +media-query($bp.xxs)
         span:first-child
           margin-left: 0
diff --git a/pkg/webui/console/components/events/events.styl b/pkg/webui/console/components/events/events.styl
index 077645d17..d9d7b5ba4 100644
--- a/pkg/webui/console/components/events/events.styl
+++ b/pkg/webui/console/components/events/events.styl
@@ -48,7 +48,7 @@ $event-container-height = 40px
     font-weight: $fw.bold
     +media-query($bp.s)
         display: none
-  
+
   .cell-data
     +media-query($bp.l)
         display: none
@@ -70,14 +70,17 @@ $event-container-height = 40px
   flex-grow: 1
   right: $cs.s
   display: flex
-  justify-content: flex-end
   +media-query-min($bp.xs)
+    justify-content: flex-end
     position: sticky
 
   .actions
     background-color: white
     display: flex
 
+    button
+      margin-right: 0 !important
+
 .toggle-container
   display: flex
   align-items: center
```

**Pull request merged**